pennylane/ops/op_math/pow.py

```diff
@@ -341,8 +342,17 @@ def generator(self):
     def pow(self, z):
         return [Pow(base=self.base, z=self.z * z)]
 
+    # pylint: disable=arguments-renamed, invalid-overridden-method
+    @property
+    def has_adjoint(self):
+        return isinstance(self.z, int)
+
     def adjoint(self):
-        return Pow(base=qml.adjoint(self.base), z=self.z)
+        if isinstance(self.z, int):
```

Comment:
Similarly here as above:
```suggestion
        if isinstance(self.z, int) or float(self.z).is_integer() :
```